cmd/oras/internal/option/remote.go

```diff
@@ -109,21 +151,42 @@ func (opts *Remote) tlsConfig() (*tls.Config, error) {
 	return config, nil
 }
 
+var defaultDialer = &net.Dialer{
+	Timeout:   30 * time.Second,
+	KeepAlive: 30 * time.Second,
+}
+
+// DialContext connects to the addr on the named network using
+// the provided context.
+func (opts *Remote) DialContext(ctx context.Context, network, addr string) (net.Conn, error) {
+	var matched *ResolveEntry
+	for _, r := range opts.Resolves {
+		if addr == fmt.Sprintf("%s:%d", r.from, r.port) {
+			matched = r
+			break
+		}
+	}
+	if matched == nil {
+		return defaultDialer.DialContext(ctx, network, addr)
+	}
+	return net.DialTCP(network, nil, &net.TCPAddr{IP: matched.to, Port: matched.port})
+}
```

Comment:
The `net.DialTCP` does not accept a context. How about
```go
type resolveDialer struct {
	base   *net.Dialer
	resolve map[string]string
}

func (rd *resolveDialer) DialContext(ctx context.Context, network, addr string) (net.Conn, error) {
	if resolved, ok := rd.resolve[addr]; ok {
		addr = resolved
	}
	return rd.base.DialContext(ctx, network, addr)
}
```
Will above code work?

Reply:
It surely works, but using mapping surely reduces extensibility for supporting ipv6. Resolved accordingly.

Reply:
Checked it's non-relavant.